string graph_filename
---
bool success_flag